Python3 support

quarnster opened the pull request:
FYI in case any one is interested. Works on Go 1.1-ish as it's not been released yet, but "go version devel +b27b1ff18f39 Wed Apr 10 07:15:49 2013 +0200 darwin/amd64". Also tested on darwin/386 and linux/amd64. Other platforms untested.

And if you're not interested in Python 3, there's also a memory leak fix in https://github.com/quarnster/gopy/commit/8649515062388d59ae4be3b9f3f056a9170eb396 that should be applied to all versions.

quarnster committed:
Unit testing wip

quarnster committed:
Fixed method calling

quarnster committed:
Adjusted for lime

quarnster committed:
Use c malloced pointer for function references

quarnster committed:
removed gopy.test binary

quarnster committed:
python3 port

quarnster committed:
Added SetInterrupt + test

quarnster committed:
Fix memory leaks in the variable argument BaseObject.Call* methods

quarnster committed:
Count how many times the goroutine has been locked to the os thread.
Otherwise we prematurely release the goroutine from the os thread
resulting in a crash, as the GIL locking mechanisms allow reentrance.

adzenith committed:
Fixes for "function declaration isn't a prototype"

quarnster committed:
Merge pull request #1 from adzenith/prototype-fix

Fixes for "function declaration isn't a prototype"

charlesbalderson commented:
Is this project dead? Would be brilliant to get these changes merged!

qur commented:
In the sense that I am still interested in it, and want to work on it - no.  It's just not managed to make it high enough up the list of "things to do" for a while ... :(

I've cherry-picked most of the fixes from here.  However the Python 3 changes are not backwards compatible - and I still need 2.7 compatibility ... needs some more thought.

quarnster commented:
Note that [python 3 can make a call](http://hg.python.org/cpython/file/e0f86c3b3685/Modules/faulthandler.c#l1093) to sigaltstack [even if](http://hg.python.org/cpython/file/e0f86c3b3685/Python/pythonrun.c#l367) it's [requested](http://hg.python.org/cpython/file/e0f86c3b3685/Python/pythonrun.c#l375) that signal handler's aren't installed.

C libraries messing with the signal handlers [is not supported](https://code.google.com/p/go/issues/detail?id=4216) and [will cause problems](https://code.google.com/p/go/issues/detail?id=5287). This is true for both Python2 and 3, but 3's a bit more problematic as there's no runtime choice as to whether sigaltstack is called or not.

adzenith committed:
Fix for "function is not needed and will not be emitted" errors

quarnster committed:
Merge pull request #2 from adzenith/python3

Fix for "function is not needed and will not be emitted" errors

quarnster committed:
Remove SetInterrupt as it requires signals.
Panic when trying to use signals
(https://code.google.com/p/go/issues/detail?id=5287)
Fix some test flakiness.

l1n3n01z committed:
Using exposed function instead of macro, to reduce dependency on Python internals

Commit:
Fixing typo; Changing name of calling function

Commit:
Windows compiler settings in cgo.go

Commit:
test for basic handling of PyLong_type

Commit:
Handling type determination correctly for windows

Due to issue https://code.google.com/p/go/issues/detail?id=4339 in cgo,
we had to change how we determine types of python objects. We are unable
to compare to static pointers declared in the python DLL.

Only one test still failing

Commit:
Removed hard coded test failure

quarnster committed:
Get rid of unneeded libffi dependency

Commit:
Merging upstream

Merging changes from upstream, removing dependency on libffi

Commit:
added newline for c compliance

Commit:
added newline for c compliance

quarnster committed:
Merge pull request #4 from l1n3n01z/windows_fix

Windows fix

quarnster committed:
Windows mingw workarounds

quarnster committed:
_PyObject -> GoPyObject where it makes sense
